streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -633,6 +634,43 @@ public final void copartitionSources(final Collection<String> sourceNodes) {
         copartitionSourceGroups.add(Collections.unmodifiableSet(new HashSet<>(sourceNodes)));
     }
 
+    public void validateCopartition() {
+        final List<Set<String>> copartitionGroups =
```

Comment:
If I understand this code correct, `copartitionGroups` take the list of co-partitioned nodes (ie, processor names -> `copartitionSourceGroups`) and replaces each processor name (each processor should be a source-processor?) with the corresponding source topic name?

If yes, a comment might be helpful. Also, name rename `copartitionGroups` -> `allCopartitionedSourceTopics` ?

Reply:
Makes total sense. thanks for your advice.